### ISSUE TITLE
Unnecessary switching to intermediate 48hz rate before going to 96khz

### ISSUE THREAD

**Issue by geofstro** [vincentneo/LosslessSwitcher#134]
Whenever Apple Music switches from 44.1khz to 96khz music, instead of switching directly to 96khz an intermediate 48khz is first applied.

This is causing problems for my own app, which is watching for sample rate changes on the output device, and making decisions based on those.

Can this intermediate switching somehow be avoided?

**Comment by vincentneo** [comment]
The problem is due to Apple Music sometimes logging lower resolution values, which LosslessSwitcher ends up picking up.

As such, this issue remains unsolvable at present.

**Comment by geofstro** [comment]
I agree with you; this is something Apple is doing,and it seems to do it consistently.

Whenever the sample rate needs to change from the current rate to another rate it always changes first to 48khz, then after a second or two it changes to the correct rate.

So I believe it should be possible for you to wait for that short period,to see if it changes a second time, before changing to
the rate on the device?

The only time this won't apply is when the actual sample rate should  be 48khz; but then, after the short wait, it will still be at 48khz.

**Comment by vincentneo** [comment]
> So I believe it should be possible for you to wait for that short period,to see if it changes a second time, before changing to
the rate on the device?

I can consider trying this

**Comment by geofstro** [comment]
That would be great! Thanks.

**Comment by iammohitsakhuja** [comment]
Not sure if I should open a separate issue, but the issue I am facing might be a subset of the above issue mentioned - wherein the sample rate gets set to 96 kHz, but then switches back to 48 kHz. This happens when the song is played from anywhere in Apple Music except Library. When it is played from Library, this issue does not occur. The CPU usage for Lossless Switcher also becomes 60-100% for a few seconds when this happens.

Build version: `1.1.1-beta1`
macOS version: `Sequoia 15.0`
Mac: `Macbook Pro M1`
Example song: `https://music.apple.com/in/album/fun-feat-tove-lo/1053933969?i=1053934858`

**Comment by cbowman54** [comment]
What rate is applied when audio signal is a higher Khz than the maximum recognized by the default DAC
